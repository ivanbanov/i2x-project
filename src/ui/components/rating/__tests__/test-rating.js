import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Rating from '../';

jest.mock('../styles.styl', () => ({}));
jest.mock('src/ui/components/icon', () => (() => <div>Icon</div>));

describe('Button', () => {
  it('should render properly', () => {
    const component = renderer.create(
      <Rating rating={5} readonly />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const fn = jest.fn();

    const component = shallow(
      <Rating onClick={fn} />
    );

    component.find('div > div').first().simulate('click');

    expect(fn).toHaveBeenCalled();
  });
});
