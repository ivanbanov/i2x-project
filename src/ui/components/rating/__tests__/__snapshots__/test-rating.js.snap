// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
<div
  className="undefined"
>
  <div
    className="undefined"
    onClick={[Function]}
  >
    <div>
      Icon
    </div>
  </div>
  <div
    className="undefined"
    onClick={[Function]}
  >
    <div>
      Icon
    </div>
  </div>
  <div
    className="undefined"
    onClick={[Function]}
  >
    <div>
      Icon
    </div>
  </div>
  <div
    className="undefined"
    onClick={[Function]}
  >
    <div>
      Icon
    </div>
  </div>
  <div
    className="undefined"
    onClick={[Function]}
  >
    <div>
      Icon
    </div>
  </div>
</div>
`;
